src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -342,18 +350,30 @@ private boolean modifyUserFullName(User user, SAML2Profile saml2Profile) {
      * @param emails user emails.
      * @return true if the current user is modified.
      */
-    private boolean modifyUserEmail(User user, List<?> emails) {
+    private boolean modifyUserEmail(User user, List<String> emails) {
         String userEmail = null;
         boolean saveUser = false;
-        if (emails != null && !emails.isEmpty()) {
-            userEmail = (String) emails.get(0);
+        if (emails == null || emails.isEmpty()) {
+            LOG.warning("There is not Email attribute '" + getEmailAttributeName() + "' for user : " + user.getId());
+            return saveUser;
+        }
+
+        for (String item : emails) {
+            if (StringUtils.isNotEmpty(item)) {
+                userEmail = item;
+                break;
+            }
+        }
+
+        if (StringUtils.isBlank(userEmail)) {
+            LOG.warning("The Email is blank for user : " + user.getId());
         }
 
         try {
             if (user != null && StringUtils.isNotBlank(userEmail)) {
                 UserProperty currentUserEmailProperty = user.getProperty(UserProperty.class);
-                if (currentUserEmailProperty != null
-                        && userEmail.compareTo(StringUtils.defaultIfBlank(currentUserEmailProperty.getAddress(), "")) != 0) {
+                if (currentUserEmailProperty == null ||
```

Comment:
This is changing the behavior, now the property will be set even if `currentUserEmailProperty` is null, is that what you want to achieve?